### DISCUSSION TITLE
Impossible to use Renovatebot with Gemfury private repositories

### DISCUSSION THREAD

**Discussion by carlos-jt** [renovatebot/renovate#26318] [Request Help]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I have not seen this working

### Describe the problem

Hello,

I am not able to make it work Renovate with Gemfury npm private repositories.

I have this configuration:
```
  "hostRules": [
    {
      "hostType": "npm",
      "baseUrl": "https://npm.fury.io/<redacted>",
      "token": "${{secrets.RUBYGEMS_SERVER_GEM_FURY_IO_TOKEN}}"
    },
```

But I'm getting this error:
```
DEBUG: GET https://npm.fury.io/<redacted>/@<redacted>%2F<redacted> = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=401 retryCount=0, duration=387)
DEBUG: Failed to look up npm package @<redacted>/<redacted>
{
  "dependency": "@<redacted>/<redacted>"
  "packageFile": "package.json"
}
```
This is the `.npmrc` that I'm using:

```
always-auth=true
//npm.fury.io/<redacted>/:_authToken=${BUNDLE_GEM__FURY__IO}
@<redacted>:registry=https://npm.fury.io/<redacted>/
```

Because the `authToken` reference here is different, I tried to pass the same `npmrc` in the Renovate configuration but pointed to the same secret that I'm using in `hostRules` but without success. 

I tried different configurations:
- Using `npmToken` instead of token
- Using fresh encrypted token directly instead of GitHub secret
- Create a new token
- Play with different `npmrc`

If i use this token locally, i can authenticate.

Could someone help? thank you

### Relevant debug logs

<details><summary>Logs</summary>

```

DEBUG: http statistics
{
  "urls": {
    "https://api.github.com/graphql (POST,200)": 4,
    "https://api.github.com/repos/<redacted>/<redacted>/contents/.github/renovate.json (GET,404)": 1,
    "https://api.github.com/repos/<redacted>/<redacted>/pulls (GET,200)": 27,
    "https://npm.fury.io/<redacted>/@<redacted>%2F<redacted> (GET,401)": 1,
    "https://npm.fury.io/<redacted>/@<redacted>%2F<redacted> (GET,401)": 1,
    "https://registry.npmjs.org/rollup-plugin-dts (GET,304)": 1

```

</details>

### Have you created a minimal reproduction repository?

I have linked to a minimal reproduction in the description above

**Comment by rarkins** [comment]
Take one of the URLs which fails with the 401, and `curl` it manually. Tell us here the exact curl command you run (redact the token value obviously).

**Reply by carlos-jt** [reply]
Hello:

If i try it like this, it works:

```
curl -I https://<TOKEN>@npm.fury.io/<org>/@<org>/<artifact>
HTTP/1.1 200 OK
Server: Cowboy
Nel: {"report_to":"heroku-nel","max_age":3600,"success_fraction":0.005,"failure_fraction":0.05,"response_headers":["Via"]}
Connection: keep-alive
Content-Type: application/json; charset=utf-8
Etag: "8f790f64366473cfe80e6dad23b94150"
Last-Modified: Fri, 15 Dec 2023 12:06:49 GMT
Vary: Accept-Encoding
Date: Fri, 15 Dec 2023 14:28:45 GMT
Via: 1.1 vegur
```

Thank you!

**Reply by rarkins** [reply]
Can you find a way to achieve that with the token in a _header_? I'm trying to understand if it's a Bearer or Basic token, and then troubleshoot accordingly

**Reply by carlos-jt** [reply]
I tried to find in all Gemfury docs but the only ways that i found to achieve it were:
- Using token in the URL (like i posted or in the path)
- Using the token in the `.npmrc` like `//npm-proxy.fury.io/:_authToken=${FURY_TOKEN}`

**Reply by carlos-jt** [reply]
Good morning @rarkins,

Do you have any other suggestions that can i follow to fix that issue? i'm re-reading all documentation but i don't find any suitable fix for doing that.

**Reply by rarkins** [reply]
Unfortunately this would require a reproduction repository to debug further
